src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -1293,12 +1353,34 @@ public static void dispatchMutationsWithRetryOnDifferentSystem(List<? extends IM
                     if (accordResult != null)
                     {
                         IAccordService accord = AccordService.instance();
-                        TxnResult.Kind kind = accord.getTxnResult(accordResult, true, consistencyLevel, requestTime).kind();
-                        if (kind == retry_new_protocol)
+                        TxnResult.Kind kind = accord.getTxnResult(accordResult).kind();
+                        if (kind == retry_new_protocol && failure == null)
+                        {
+                            Tracing.trace("Accord returned retry new protocol");
+                            logger.debug("Retrying mutations on different system because some mutations were misrouted according to Accord");
                             continue;
+                        }
                         Tracing.trace("Successfully wrote Accord mutations");
                     }
                 }
+                catch (TopologyMismatch e)
+                {
+                    // Don't suppress existing failure
+                    if (failure == null)
+                    {
+                        // For now assuming topology mismatch is caused by a race misrouting
+                        Tracing.trace("Accord returned topology mismatch, retrying: " + e.getMessage());
+                        logger.debug("Accord returned topology mismatch, retrying: " + e.getMessage());
+                        continue;
```

Comment:
this feels very dangerous without a retry limit....

```
insert into tbl ... IF NOT EXISTS;
drop table tbl;
```

this causes a race condition where we infinite loop blocking this thread for life....

`TopologyMismatch` was never expected to be a retryable event, it was a die-in-a-fire event.